diffusion/datasets/image.py

```diff
@@ -0,0 +1,150 @@
+        streaming_kwargs.setdefault('shuffle_block_size', 1 << 18)
+        streaming_kwargs.setdefault('shuffle_algo', 'py1s')
+
+        super().__init__(
+            streams=streams,
+            remote=remote,
+            local=local,
+            **streaming_kwargs,
+        )
+
+        self.transform = transform
+        self.image_key = image_key
+        self.image_output_key = image_output_key if image_output_key is not None else image_key
+        self.return_all_fields = return_all_fields
+
+    def __getitem__(self, index):
+        sample = super().__getitem__(index)
+        # Image
+        if not isinstance(sample[self.image_key], Image.Image):
+            img = Image.open(BytesIO(sample[self.image_key])).copy()
+        else:
+            img = sample[self.image_key].copy()
+        if img.mode != 'RGB':
+            img = img.convert('RGB')
+        # Image transforms
+        if self.transform is not None:
+            img = self.transform(img)
+        # Make the output
+        if self.image_output_key != self.image_key and self.image_output_key in sample:
+            raise ValueError(f'Output key {self.image_output_key} already exists in the sample.')
+        if self.return_all_fields:
+            output = {**sample, self.image_output_key: img}
+        else:
+            output = {self.image_output_key: img}
+        return output
+
+
+def build_streaming_image_dataloader(
+    remote: Union[str, List],
+    local: Union[str, List],
+    batch_size: int,
+    transform: Optional[List[Callable]] = None,
+    image_key: str = 'image',
+    streaming_kwargs: Optional[Dict] = None,
+    dataloader_kwargs: Optional[Dict] = None,
+):
+    """Builds a streaming LAION dataloader.
+
+    Args:
+        remote (str, Sequence[str]): One or more remote directories (S3 or local filesystem) where dataset is stored.
+        local (str, Sequence[str]): One or more local filesystem directories where dataset is cached during operation.
+        batch_size (int): The batch size to use for both the ``StreamingDataset`` and ``DataLoader``.
+        transform (Optional[Callable]): The transforms to apply to the image. Default: ``None``.
+        image_key (str): Key associated with the image in the streaming dataset. Default: ``'image'``.
+        streaming_kwargs (dict, optional): Additional arguments to pass to the ``StreamingDataset``. Default: ``None``.
+        dataloader_kwargs (dict, optional): Additional arguments to pass to the ``DataLoader``. Default: ``None``.
+    """
+    # Handle ``None`` kwargs
+    if streaming_kwargs is None:
+        streaming_kwargs = {}
+    if dataloader_kwargs is None:
+        dataloader_kwargs = {}
+
+    # Check types for remote and local
+    if isinstance(remote, str) and isinstance(local, str):
+        remote, local = [remote], [local]
+    elif isinstance(remote, Sequence) and isinstance(local, Sequence):
+        if len(remote) != len(local):
+            ValueError(
```

Comment:
This is suppose to raise ValueErrors here, not just instantiate them, right?

Reply:
This is to allow passing in a single `remote` + `local` pair as a string instead of a list of multiple, so it shouldn't throw an error. Though questionable if this is how we'd like this implemented.

Reply:
I meant:
```suggestion
            raise ValueError(
```

Reply:
Oops! Yes you're right. Nice catch!